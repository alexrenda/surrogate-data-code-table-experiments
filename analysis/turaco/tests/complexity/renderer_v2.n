fun (
  rimLight[4], isCelShadingEnabled[2], sunPositionBase[2],
  gamma[2], worldNormal[3], ssao[3],
  diffuseColor[4],
  diffuse[4], specular[4],
  emissionBase[3], isWater[2], isParticle[2]
) {
  sunPosition = sin(sunPositionBase[0] * pi / 180);

  sunMixFactor = 0.5 - sunPosition / 2;
  ambientCoolBase = exp(log1b([0.302, 0.451, 0.471]) * gamma[0]);
  ambientWarmBase = exp(log1b([0.765, 0.573, 0.400]) * gamma[0]);
  // ambientCoolBase = exp(gamma[0] * [-1.1973282616, -0.7962879395, -0.752897185]);
  // ambientWarmBase = exp(gamma[0] * [-0.2678794452, -0.5568695623, -0.9162907319]);

  if (0.5 > sunMixFactor) {
    ambientCool = ambientCoolBase / 2;
    ambientWarm = ambientWarmBase / 2;
  } else {
    ambientCool = ambientCoolBase * sunMixFactor;
    ambientWarm = ambientWarmBase * sunMixFactor;
  }

  if (0 > sunMixFactor) {
    skyLight = ambientCool;
    groundLight = ambientWarm;
  } else {
    if (sunMixFactor > 1) {
      skyLight = ambientWarm;
      groundLight = ambientCool;
    } else {
      skyLight = ambientCool * (1 - sunMixFactor) + ambientWarm * sunMixFactor;
      groundLight = ambientWarm * (1 - sunMixFactor) + ambientCool * sunMixFactor;
    }
  }

  worldNormalMixFactor = (1.0 + worldNormal[2]) / 2;
  ambientLight = groundLight * (1 - worldNormalMixFactor) + skyLight * worldNormalMixFactor;
  ambient = ambientLight * ssao * [diffuseColor[0], diffuseColor[1], diffuseColor[2]];

  if (0.00316228 > sunPosition) {
    emission = emissionBase * 0.1;
  } else {
    // assert: 0.00316228 < sunPosition < 1
    sunPositionPow = exp(log1b(sunPosition) * 0.4);
    emission = emissionBase * sunPositionPow;
  }

  out0rgb = [ambient[0], ambient[1], ambient[2]] + [diffuse[0], diffuse[1], diffuse[2]] + [rimLight[0], rimLight[1], rimLight[2]] + emission;

  if (isWater[0] > 0) {
    out0a = 0;
  } else {
    out0a = diffuseColor[3];
  }

  out1a = diffuseColor[3];

  if (isParticle[0] > 0) {
    out1rgb = [0,0,0];
  } else {
    out1rgb = [specular[0], specular[1], specular[2]];
  }

  out = [out0rgb[0], out0rgb[1], out0rgb[2], out0a, out1rgb[0], out1rgb[1], out1rgb[2], out1a];
  return out;
}
