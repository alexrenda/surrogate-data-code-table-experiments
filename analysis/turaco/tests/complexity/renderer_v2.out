lllll 1826.4242757807983
llllr 2091.8443879563233
lllrl 1912.8976465059732
lllrr 2184.317758681498
llrll 2068.3719689983564
llrlr 2350.2481146834284
llrrl 2160.330684226714
llrrr 2448.206829911786
lrllll 1333.6727806269596
lrlllr 1561.7896902952245
lrllrl 1407.7117505163812
lrllrr 1641.8286601846462
lrlrll 1541.5169877071344
lrlrlr 1786.0899308849469
lrlrrl 1621.0413020997385
lrlrrr 1871.614245277551
lrrlll 5746.324644064372
lrrllr 6210.151739987995
lrrlrl 5898.933676038912
lrrlrr 6368.760771962537
lrrrll 6169.660780231206
lrrrlr 6649.943909664377
lrrrrl 6327.75515670893
lrrrrr 6814.038286142101
rllll 1896.1057456301855
rlllr 2166.371523800015
rllrl 1984.194338353462
rllrr 2260.4601165232916
rlrll 2142.483463402097
rlrlr 2429.2052750814746
rlrrl 2236.0574006285565
rlrrr 2528.7792123079334
rrllll 1383.1976222121314
rrlllr 1615.3458115384997
rrllrl 1458.5803519875874
rrllrr 1696.7285413139557
rrlrll 1594.7273222445344
rrlrlr 1843.3315450804503
rrlrrl 1675.5953965231731
rrlrrr 1930.199619359089
rrrlll 5985.783662236817
rrrllr 6458.990740856397
rrrlrl 6141.51935511001
rrrlrr 6620.72643372959
rrrrll 6417.695204490518
rrrrlr 6907.358316619646
rrrrrl 6578.916241866894
rrrrrr 7074.579353996022
